azerothcore/azerothcore-wotlk
# fix(DB/SAI): Steam Pump Overseer toughen not stacking.

On Just Summoned SAI is used for #13618 (NYI)

## Changes Proposed:
- Add knockdown proc to creature_template_addon
- Make toughen add 1 stack for every 10% missing health.
- Add no_exp extra flag 

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14440
- Closes https://github.com/chromiecraft/chromiecraft/issues/4692

## SOURCE:

https://youtu.be/79XhAIFv9L8?t=244
https://youtu.be/SohyP4uZucY?t=187 - no exp
cmangos
## Tests Performed:

- tested

## How to Test the Changes:


.go c id 18340
confirm knockdown happens
Use gm spells (.learn all gm) - dmg1-3
Push him slowly from 100% hp to 10% hp, confirm that the npc gains 1x stack every 10% missing health, up to 9 stacks at or under 10% health.
Kill it and see that it doesn't give any exp.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
